executor/src/main/java/org/k8loud/executor/openstack/OpenstackClientProvider.java

```diff
@@ -0,0 +1,95 @@
+package org.k8loud.executor.openstack;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import org.k8loud.executor.exception.OpenstackException;
+import org.openstack4j.api.OSClient;
+import org.openstack4j.api.client.IOSClientBuilder;
+import org.openstack4j.api.exceptions.AuthenticationException;
+import org.openstack4j.api.exceptions.ConnectionException;
+import org.openstack4j.core.transport.Config;
+import org.openstack4j.model.common.Identifier;
+import org.openstack4j.model.identity.v3.Token;
+import org.openstack4j.openstack.OSFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+
+import static org.k8loud.executor.exception.code.OpenstackExceptionCode.AUTHENTICATION_ERROR;
+import static org.openstack4j.api.OSClient.OSClientV3;
+
+@RequiredArgsConstructor
+@Getter
+@Slf4j
+public class OpenstackClientProvider {
+    @Autowired
+    private final OpenstackProperties openstackProperties;
+    private Token token = null;
+
+    public boolean isTokenActive() {
+        if (token == null) {
+            log.debug("Token is not set");
+            return false;
+        }
+
+        return isTokenExpiringSoon(token);
```

Comment:
What about case where token is set and isn't expiring soon, shouldn't it be active too?

Reply:
Yes good catch, will return the opposite of isTokenExpiring